healthcheck/generalchecks/src/main/java/org/apache/felix/hc/generalchecks/BundlesStartedCheck.java

```diff
@@ -79,78 +82,85 @@ protected void activate(BundleContext bundleContext, Config config) {
-            }
+    private boolean isNotFragment(Bundle bundle) {
+        boolean isFragment = StringUtils.isNotBlank(bundle.getHeaders().get(Constants.FRAGMENT_HOST));
+        if (isFragment) {
+            LOG.debug("Ignoring bundle fragment: {}", bundle.getSymbolicName());
         }
-
-        String includeMsg = !includesRegex.pattern().equals(".*") ? " for pattern " + includesRegex.pattern(): "";
-        String baseMsg = relevantBundlesCount + " bundles" + includeMsg;
-        String excludedMsg = countExcluded > 0 ? " (" + countExcluded + " excluded via pattern " + excludesRegex.get().pattern() + ")" : "";
-        if (inactiveCount > 0) {
-            log.info("Found  " + inactiveCount + " inactive of " + baseMsg + excludedMsg);
-        } else {
-            log.info("All " + baseMsg + " are started" + excludedMsg);
+        return !isFragment;
+    }
+    
+    private boolean isNotLazyStarting(Bundle bundle) {
+        // support lazy activation (https://www.osgi.org/developer/design/lazy-start/)
+        boolean isLazyStarting = bundle.getState() == Bundle.STARTING && isLazyActivation(bundle);
+        if (isLazyStarting) {
+            LOG.debug("Ignoring lazily activated bundle {}", bundle.getSymbolicName());
         }
+        return !isLazyStarting;
+    }
 
-        return new Result(log);
+    private boolean isIncluded(Bundle bundle) {
+        boolean matches = includesRegex.matcher(bundle.getSymbolicName()).matches();
+        if (matches) {
+            LOG.debug("Bundle {} not matched by {}", bundle.getSymbolicName(), includesRegex);
+        }
+        return matches;
+    }
+    
+    private boolean isNotExcluded(Bundle bundle) {
+        boolean matches = excludesRegex.isPresent() ? excludesRegex.get().matcher(bundle.getSymbolicName()).matches() : false;
+        if (matches) {
+            LOG.debug("Bundle {} excluded {}", bundle.getSymbolicName(), excludesRegex);
+        }
+        return !matches;
     }
 
-    private static boolean isLazyActivation(Bundle b) {
+    private boolean isLazyActivation(Bundle b) {
         return Constants.ACTIVATION_LAZY.equals(b.getHeaders().get(Constants.BUNDLE_ACTIVATIONPOLICY));
     }
 
-    private static String getStateLabel(int state) {
+    private boolean isActive(FormattingResultLog log, Bundle bundle) {
+        boolean isActive = bundle.getState() == Bundle.ACTIVE;
+        String bundleInfo = getBundleInfo(bundle);
+        if (isActive) {
+            log.debug(bundleInfo);
+        } else {
+            String msg = "Inactive " + bundleInfo;
+            if (useCriticalForInactive) {
+                log.critical(msg);
+            } else {
+                log.warn(msg);
+            }
+        }
+        return isActive;
+    }
+
+    private String getBundleInfo(Bundle bundle) {
+        return String.format("bundle with id:%d, name:%s, state:%s)", bundle.getBundleId(), 
```

Comment:
Nitpick: the current format is just `Bundle {} {}: {}` (without labelling `id`/`name`/`state` on every line)
I think the values are self-explanatory in this context, I would keep it as is